pennylane/devices/qubit/apply_operation.py

```diff
@@ -228,9 +229,27 @@ def _(op: type_op, state):
         [1., 0.]], requires_grad=True)
 
     """
+    if sp.sparse.issparse(op.matrix()):
```

Comment:
Isn't `op.matrix()` always dense? How would we hit the this branch?

Reply:
@albi3ro If `op` is a `QubitUnitary` initialized by sparse matrix, it's `matrix()` will return a CSR

Reply:
Could we instead do `if op.has_sparse_matrix and not op.has_matrix`?

Reply:
Fixed in extra PR https://github.com/PennyLaneAI/pennylane/pull/6986